vortex-scalar/src/primitive.rs

```diff
@@ -270,3 +272,47 @@ impl From<usize> for Scalar {
         Scalar::primitive(value as u64, Nullability::NonNullable)
     }
 }
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum NumericOperator {
+    Add,
+    Sub,
+    Mul,
+    Div,
+    // Missing from arrow-rs:
+    // Min,
+    // Max,
+    // Sqrt,
+    // Pow,
+    // In IEEE-754 operations involving NaN are non-commutative wrt NaN payload/quietness
+    // RAdd,
+    // RSub,
+    // RMul,
+    // RDiv,
+}
+
+impl PrimitiveScalar<'_> {
+    pub fn numeric_operator(
```

Comment:
Can we call this `checked_numeric_operator` and return `VortexResult<Option<Scalar>>`, then use checked_add, etc from num-traits to perform the operators.